tests/unit/test_groups.py

```diff
@@ -0,0 +1,527 @@
+
+        task = kitipy.Task(name='foobar')
+        task.callback = mock.Mock()
+
+        click_ctx = click.Context(task)
+        mock_get_current_click_ctx.return_value = click_ctx
+
+        task.invoke(click_ctx)
+
+        task.callback.assert_called()
+
+
+def test_group_is_disabled_when_one_of_its_filter_is_negative(click_ctx):
+    filter = mock.Mock(return_value=False)
+    group = kitipy.Group(name='foobar', filters=[filter])
+
+    assert group.is_enabled(click_ctx) == False
+    filter.assert_called()
+
+
+def test_group_is_disabled_when_it_is_hidden(click_ctx):
+    filter = mock.Mock(return_value=True)
+    group = kitipy.Group(name='foobar', filters=[filter])
+    group.hidden = True
+
+    assert group.is_enabled(click_ctx) == False
+    filter.assert_called()
+
+
+def test_group_is_enabled(click_ctx):
+    filter = mock.Mock(return_value=True)
+    group = kitipy.Group(name='foobar', filters=[filter])
+
+    assert group.is_enabled(click_ctx) == True
+    filter.assert_called()
+
+
+def test_invoke_disabled_group_raises_an_exception(click_ctx):
+    group = kitipy.Group(name='foobar')
+    group.hidden = True
+
+    with pytest.raises(kitipy.TaskError):
+        group.invoke(click_ctx)
+
+
+def test_group_invoke_updates_the_executor_basedir_using_the_group_cwd(
+        kctx, executor):
+    with mock.patch('click.get_current_context') as mock_get_current_click_ctx:
+        group = kitipy.Group(name='foobar',
+                             cwd='some/base/dir',
+                             callback=mock.Mock(),
+                             invoke_without_command=True)
+
+        click_ctx = click.Context(group, obj=kctx)
+        mock_get_current_click_ctx.return_value = click_ctx
+
+        group.invoke(click_ctx)
+
+        executor.cd.assert_called_with('some/base/dir')
+        group.callback.assert_called()
+
+
+def test_group_merging_adds_source_commands_and_transparent_groups():
+    src_foo = kitipy.Task(name='foo')
+    src_bar = click.Command(name='bar')
+    src_baz = kitipy.Group(name='baz')
+    src_acme = kitipy.Group(name='acme')
+    src = kitipy.Group(commands={
+        'foo': src_foo,
+        'bar': src_bar,
+        'baz': src_baz,
+    })
+    src.add_transparent_group(src_acme)
+
+    dest_first = kitipy.Task(name='first')
+    dest = kitipy.Group(commands={'first': dest_first})
+    dest.merge(src)
+
+    expected = {
+        'bar': src_bar,
+        'baz': src_baz,
+        'first': dest_first,
+        'foo': src_foo
+    }
+    assert dest.commands == expected
+    assert dest.transparent_groups == [src_acme]
+
+
+def test_group_get_command_looks_for_the_given_command_in_transparent_groups():
+    foo = kitipy.Task(name='foo')
+    acme = kitipy.Group(commands={'foo': foo})
+    root = kitipy.Group()
+    root.add_transparent_group(acme)
+
+    click_ctx = click.Context(root)
+    assert root.get_command(click_ctx, 'foo') is not None
```

Comment:
Can't you test that the return value of `get_command` is `foo`? Or am I entirely wrong?

Reply:
Good catch!